app/controllers/PaymentController.scala

```diff
@@ -56,7 +56,7 @@ class PaymentController @Inject()(val checkSessionTimeout: SessionTimeoutPredica
       case Some(utr) =>
         payApiConnector.startPaymentJourney(utr, paymentAmountInPence, isAgent).map {
           case model: PaymentJourneyModel => Redirect(model.nextUrl)
-          case _: PaymentJourneyErrorResponse => throw new Exception("Failed to start payments journey due to downstream response")
+          case _: PaymentJourneyResponse => throw new Exception("Failed to start payments journey due to downstream response")
```

Comment:
we should match on PaymentJourneyErrorResponse(status, message)
and provide error details in exception like, 
Exception(s"Failed to start payments journey due to downstream response: $status - $message")